### PR TITLE
[#12588] Add unit test for every method in CommentEditFormComponent

### LINKED ISSUE #12588: Improve test code coverage of core components
**Description of feature/enhancement**
The components in `src/web/app/components` are used throughout TEAMMATES, but some of them have poor test coverage. Let's improve the test coverage in these components:

**Steps to contribute**

1. Choose component(s) to work on in the table below, and comment on this issue to indicate that you'd like to contribute tests for the component(s)
2. Create a PR with tests for the component, check the codecov link to ensure that the functionality not covered are indeed tested
3. Submit the PR, link it to this issue by starting the description with "Part of #12588"

Components:
| Component | Assignee | Codecov | PR | 
|-----------|----------|---------|------|
|  AddingQuestionPanelComponent   |  @sdpunit    |  [link](https://app.codecov.io/github/TEAMMATES/teammates/blob/master/src%2Fweb%2Fapp%2Fcomponents%2Fadding-question-panel%2Fadding-question-panel.component.ts) | https://github.com/TEAMMATES/teammates/pull/12632 | 
|  CommentEditFormComponent   |  @HarryLu33   |  [link](https://app.codecov.io/github/TEAMMATES/teammates/blob/master/src%2Fweb%2Fapp%2Fcomponents%2Fcomment-box%2Fcomment-edit-form%2Fcomment-edit-form.component.ts) | https://github.com/TEAMMATES/teammates/pull/12604 |
|  CommentRowComponent   |  @ThomasGreen123   |  [link](https://app.codecov.io/github/TEAMMATES/teammates/blob/master/src%2Fweb%2Fapp%2Fcomponents%2Fcomment-box%2Fcomment-row%2Fcomment-row.component.ts) | https://github.com/TEAMMATES/teammates/pull/12612 | 
|  CommentTableComponent  |  @u7505884  |  [link](https://app.codecov.io/github/TEAMMATES/teammates/blob/master/src%2Fweb%2Fapp%2Fcomponents%2Fcomment-box%2Fcomment-table%2Fcomment-table.component.ts) | - |
|  CommentTableModalComponent   |  @kenneySiu  |  [link](https://app.codecov.io/github/TEAMMATES/teammates/blob/master/src%2Fweb%2Fapp%2Fcomponents%2Fcomment-box%2Fcomment-table-modal%2Fcomment-table-modal.component.ts) | https://github.com/TEAMMATES/teammates/pull/12609 | 
|  CommentVisibilityControlNamePipe, CommentVisibilityTypeDescriptionPipe, CommentVisibilityTypeNamePipe, CommentVisibilityTypesJointNamePipe   |  @cedricongjh  |  [link1](https://app.codecov.io/github/TEAMMATES/teammates/blob/master/src%2Fweb%2Fapp%2Fcomponents%2Fcomment-box%2Fcomment-visibility-setting.pipe.ts), [link2](https://app.codecov.io/github/TEAMMATES/teammates/blob/master/src%2Fweb%2Fapp%2Fcomponents%2Fcomment-box%2Fcomment-visibility-setting.pipe.ts), [link3](https://app.codecov.io/github/TEAMMATES/teammates/blob/master/src%2Fweb%2Fapp%2Fcomponents%2Fcomment-box%2Fcomment-visibility-setting.pipe.ts), [link4](https://app.codecov.io/github/TEAMMATES/teammates/blob/master/src%2Fweb%2Fapp%2Fcomponents%2Fcomment-box%2Fcomment-visibility-setting.pipe.ts) | https://github.com/TEAMMATES/teammates/pull/12589 |
|  CopyCourseModalComponent   |  @cedricongjh   |  [link](https://app.codecov.io/github/TEAMMATES/teammates/blob/master/src%2Fweb%2Fapp%2Fcomponents%2Fcopy-course-modal%2Fcopy-course-modal.component.ts) | https://github.com/TEAMMATES/teammates/pull/12590 | 
| CopySessionModalComponent    |  @Astrid641   |  [link](https://app.codecov.io/github/TEAMMATES/teammates/blob/master/src%2Fweb%2Fapp%2Fcomponents%2Fcopy-session-modal%2Fcopy-session-modal.component.ts) | https://github.com/TEAMMATES/teammates/pull/12616 |
|  CourseEditFormComponent   |  @lunoos   |  [link](https://app.codecov.io/github/TEAMMATES/teammates/blob/master/src%2Fweb%2Fapp%2Fcomponents%2Fcourse-edit-form%2Fcourse-edit-form.component.ts) | https://github.com/TEAMMATES/teammates/pull/12597 | 
|   DatepickerComponent  |  @dlimyy   |  [link](https://app.codecov.io/github/TEAMMATES/teammates/blob/master/src%2Fweb%2Fapp%2Fcomponents%2Fdatepicker%2Fdatepicker.component.ts) | https://github.com/TEAMMATES/teammates/pull/12592 |
|  DatePickerFormatter   |  @dlimyy  |  [link](https://app.codecov.io/github/TEAMMATES/teammates/blob/master/src%2Fweb%2Fapp%2Fcomponents%2Fdatepicker%2Fdatepicker-formatter.ts) | https://github.com/TEAMMATES/teammates/pull/12593 | 
| ErrorReportComponent    |  @u7488099   |  [link](https://app.codecov.io/github/TEAMMATES/teammates/blob/master/src%2Fweb%2Fapp%2Fcomponents%2Ferror-report%2Ferror-report.component.ts) | https://github.com/TEAMMATES/teammates/pull/12610 |
|  ExtensionConfirmModalComponent  |  @MatheuMa   |  [link](https://app.codecov.io/github/TEAMMATES/teammates/blob/master/src%2Fweb%2Fapp%2Fcomponents%2Fextension-confirm-modal%2Fextension-confirm-modal.component.ts) | https://github.com/TEAMMATES/teammates/pull/12622 | 
| FeedbackPathPanelComponent  |  @JYH11   |  [link](https://app.codecov.io/github/TEAMMATES/teammates/blob/master/src%2Fweb%2Fapp%2Fcomponents%2Ffeedback-path-panel%2Ffeedback-path-panel.component.ts) | https://github.com/TEAMMATES/teammates/pull/12613 | 
| LogsHistogramComponent  |  @Miranda-lrx    |  [link](https://app.codecov.io/github/TEAMMATES/teammates/blob/master/src%2Fweb%2Fapp%2Fcomponents%2Flogs-histogram%2Flogs-histogram.component.ts) | - | 
| LogsTableComponent  |  @sdpunit    |  [link](https://app.codecov.io/github/TEAMMATES/teammates/blob/master/src%2Fweb%2Fapp%2Fcomponents%2Flogs-table%2Flogs-table.component.ts) | - | 
|  LogsDetails |  @nikkixiong   |  [link](https://app.codecov.io/github/TEAMMATES/teammates/tree/master/src%2Fweb%2Fapp%2Fcomponents%2Flogs-table%2Flog-details) | https://github.com/TEAMMATES/teammates/pull/12618 | 
|  LogsLine |  @090997    |  [link](https://app.codecov.io/github/TEAMMATES/teammates/tree/master/src%2Fweb%2Fapp%2Fcomponents%2Flogs-table%2Flog-line) | https://github.com/TEAMMATES/teammates/pull/12628 | 
| QuestionEditForm|  -  |  [link](https://app.codecov.io/github/TEAMMATES/teammates/blob/master/src%2Fweb%2Fapp%2Fcomponents%2Fquestion-edit-form%2Fquestion-edit-form.component.ts) | - | 
| QuestionResponsePanelComponent |  -  |  [link](https://app.codecov.io/github/TEAMMATES/teammates/blob/master/src%2Fweb%2Fapp%2Fcomponents%2Fquestion-response-panel%2Fquestion-response-panel.component.ts) | - | 
| GqrRqgViewResponsesComponent |  -  |  [link](https://app.codecov.io/github/TEAMMATES/teammates/tree/master/src%2Fweb%2Fapp%2Fcomponents%2Fquestion-responses%2Fgqr-rqg-view-responses) | - | 
| GroupedResponsesComponent |  -  |  [link](https://app.codecov.io/github/TEAMMATES/teammates/tree/master/src%2Fweb%2Fapp%2Fcomponents%2Fquestion-responses%2Fgrouped-responses) | - | 
| GrqRgqViewResponsesComponent |  -  |  [link](https://app.codecov.io/github/TEAMMATES/teammates/tree/master/src%2Fweb%2Fapp%2Fcomponents%2Fquestion-responses%2Fgrq-rgq-view-responses) | - | 
| PerQuestionViewResponsesComponent |  -  |  [link](https://app.codecov.io/github/TEAMMATES/teammates/tree/master/src%2Fweb%2Fapp%2Fcomponents%2Fquestion-responses%2Fper-question-view-responses) | - | 
| SingleStatisticsComponent |  -  |  [link](https://app.codecov.io/github/TEAMMATES/teammates/tree/master/src%2Fweb%2Fapp%2Fcomponents%2Fquestion-responses%2Fsingle-statistics) | - | 
| InstructorResponsesViewBase |  -  |  [link](https://app.codecov.io/github/TEAMMATES/teammates/blob/master/src%2Fweb%2Fapp%2Fcomponents%2Fquestion-responses%2Finstructor-responses-view-base.ts) | - | 
| QuestionSubmissionFormComponent |  -  |  [link](https://app.codecov.io/github/TEAMMATES/teammates/blob/master/src%2Fweb%2Fapp%2Fcomponents%2Fquestion-submission-form%2Fquestion-submission-form.component.ts) | - | 
| RecipientTypeNamePipe |  @ThomasGreen123 |  [link](https://app.codecov.io/github/TEAMMATES/teammates/blob/master/src%2Fweb%2Fapp%2Fcomponents%2Fquestion-submission-form%2Frecipient-type-name.pipe.ts) | https://github.com/TEAMMATES/teammates/pull/12626 | 
| QuestionConstraint |  -  |  [link](https://app.codecov.io/github/TEAMMATES/teammates/tree/master/src%2Fweb%2Fapp%2Fcomponents%2Fquestion-types%2Fquestion-constraint) | - | 
| QuestionEditAnswerForm |  -  |  [link](https://app.codecov.io/github/TEAMMATES/teammates/tree/master/src%2Fweb%2Fapp%2Fcomponents%2Fquestion-types%2Fquestion-edit-answer-form) | - | 
| QuestionEditDetailsForm |  -  |  [link](https://app.codecov.io/github/TEAMMATES/teammates/tree/master/src%2Fweb%2Fapp%2Fcomponents%2Fquestion-types%2Fquestion-edit-details-form) | - | 
| RichTextEditorComponent |  -  |  [link](https://app.codecov.io/github/TEAMMATES/teammates/tree/master/src%2Fweb%2Fapp%2Fcomponents%2Frich-text-editor) | - | 
| SessionEditForm |  @kenneySiu  |  [link](https://app.codecov.io/github/TEAMMATES/teammates/blob/master/src%2Fweb%2Fapp%2Fcomponents%2Fsession-edit-form%2Fsession-edit-form.component.ts) | https://github.com/TEAMMATES/teammates/pull/12627 | 
| SessionsTable |  -  |  [link](https://app.codecov.io/github/TEAMMATES/teammates/tree/master/src%2Fweb%2Fapp%2Fcomponents%2Fsessions-table) | - | 
| StudentList |  -  |  [link](https://app.codecov.io/github/TEAMMATES/teammates/tree/master/src%2Fweb%2Fapp%2Fcomponents%2Fstudent-list) | - | 
| ToastComponent |  -  |  [link](https://app.codecov.io/github/TEAMMATES/teammates/tree/master/src%2Fweb%2Fapp%2Fcomponents%2Ftoast) | - | 
| ViewResultsPanelComponent |  -  |  [link](https://app.codecov.io/github/TEAMMATES/teammates/tree/master/src%2Fweb%2Fapp%2Fcomponents%2Fview-results-panel) | - | 
| VisibilityMessages |  -  |  [link](https://app.codecov.io/github/TEAMMATES/teammates/tree/master/src%2Fweb%2Fapp%2Fcomponents%2Fvisibility-messages) | - | 
| VisiblityPanel |  -  |  [link](https://app.codecov.io/github/TEAMMATES/teammates/tree/master/src%2Fweb%2Fapp%2Fcomponents%2Fvisibility-panel) | - |

### PR DESCRIPTION
**Pull request by HarryLu33**
Fixes #12588

**Outline of Solution**
Add unit test for every method in CommentEditFormComponent

### CODE REVIEW

**Comment by HarryLu33** [comment]
Ready for review

File: src/web/app/components/comment-box/comment-edit-form/comment-edit-form.component.spec.ts [hunk 1]
```diff
@@ -42,4 +44,116 @@ describe('CommentEditFormComponent', () => {
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  // toggleVisibilityTable
```
Comment (cedricongjh) [inline src/web/app/components/comment-box/comment-edit-form/comment-edit-form.component.spec.ts:48]: nit: remove the function name comments

File: src/web/app/components/comment-box/comment-edit-form/comment-edit-form.component.spec.ts [hunk 2]
```diff
@@ -42,4 +44,116 @@ describe('CommentEditFormComponent', () => {
...
+  // triggerModelChange
+  it('should trigger model change with updated field', () => {
+    const testField = 'commentText';
+    const testData = 'Updated comment text';
+
+    const emitSpy = jest.spyOn(component.modelChange, 'emit');
+
+    component.triggerModelChange(testField, testData);
+
+    // Verify that the model should be updated
```
Comment (cedricongjh) [inline src/web/app/components/comment-box/comment-edit-form/comment-edit-form.component.spec.ts:77]: nit: these verify and check comments could be removed as well, as the assertions tell us what they are checking already

File: src/web/app/components/comment-box/comment-edit-form/comment-edit-form.component.spec.ts [hunk 3]
```diff
@@ -42,4 +44,116 @@ describe('CommentEditFormComponent', () => {
...
+
+  // modifyVisibilityControl
+  it('should allow and disallow visibility control and trigger model change', () => {
+    const isAllowed = true;
+    const visibilityType = CommentVisibilityType.GIVER;
+    const visibilityControl = CommentVisibilityControl.SHOW_COMMENT;
+
+    jest.spyOn(component.visibilityStateMachine, 'allowToSee');
+    jest.spyOn(component.visibilityStateMachine, 'disallowToSee');
+    jest.spyOn(component, 'triggerModelChangeBatch');
```
Comment (cedricongjh) [inline src/web/app/components/comment-box/comment-edit-form/comment-edit-form.component.spec.ts:133]: let's keep it consistent with the rest of the spys, have them assigned to a const and use that const for `expect`

**Review by cedricongjh — COMMENTED** [review]
thank for for the PR! 

in addition to my comments, let's add a test for the ngOnInit() method, when the component is initialized with a response as well

**Comment by HarryLu33** [comment]
Changes made
Ready for review

**Review by cedricongjh — APPROVED** [review]
LGTM! thank you for your work on this!

**Review by weiquu — APPROVED** [review]
LGTM

**Pull request merged**